### PR TITLE
Support embedding an array to a mixed 2d tensor

### PR DESCRIPTION
**Pull request by bratseth**
@geirst and @jobergum fyi

### CODE REVIEW

**Review by lesters — APPROVED** [review]
Great!

File: vespajlib/src/main/java/com/yahoo/tensor/TensorType.java [hunk 1]
```diff
@@ -135,6 +142,9 @@ public static TensorType fromSpec(String specString) {
     /** The type representing the mapped subset of dimensions of this. */
     public TensorType mappedSubtype() { return mappedSubtype; }
 
+    /** The type representing the indexed subset of dimensions of this. */
+    public TensorType indexedSubtype() { return indexedSubtype; }
```
Comment (geirst) [inline vespajlib/src/main/java/com/yahoo/tensor/TensorType.java:146]: @bratseth Consider adding test for this (and mappedSubtype()) in TensorTypeTestCase.java.

**Review by geirst — COMMENTED** [review]
Great 👍

File: vespajlib/src/main/java/com/yahoo/tensor/TensorType.java (hunk 1 above)
Comment (bratseth) [reply vespajlib/src/main/java/com/yahoo/tensor/TensorType.java:146]: will do

**Comment by jobergum** [comment]
This is awesome 💯 🏅

**Pull request merged**